### DISCUSSION TITLE
Bitbucket Cloud: reduce and cache the PR lists

### DISCUSSION THREAD

**Discussion by monwolf** [renovatebot/renovate#26275] [Suggest an Idea]
### Tell us more.

I've been using renovatebot on a project for a while, this project has at least 1500 PRs closed and merged, and every time renovatebot tries to scan the project it takes more than 2 creating the PR list. 

To, improve the speed of scans on renovatebot, I would suggest using caches to save and reuse PR Lists and removing unused big fields from the request. 

To achieve this, Bitbucket Cloud provides two QS fields to filter and reduce the responses:

q can be used for adding a filter:

```
?q=updated_on > 2022-11-11T00:00:00-07:00
```

and fields to remove unused fields:

```
&fields=-values.description,-values.links,-values.summary
```

**Comment by rarkins** [comment]
Do the removed fields influence whether Bitbucket considers the PR to be "updated", or simply reduces the amount of data sent back?

**Reply by monwolf** [reply]
If I don't understand wrong the documentation [partial-response](https://developer.atlassian.com/cloud/bitbucket/rest/intro/#partial-response), it's just a cosmetic/performance tweak. Instead of removing fields, we could do an allowed list of fields to be requested.

**Reply by rarkins** [reply]
I think we know which fields we need, so that's perhaps a better alternative. And although these two topics could be separated, we should do the field reduction first before we start caching them.

**Comment by rarkins** [comment]
Converted to feature request issue: #26278
